code/modules/reagents/reagent_containers/syringes.dm

```diff
@@ -217,34 +221,57 @@
 
 
 
-/obj/item/weapon/reagent_containers/syringe/update_icon()
+/obj/item/weapon/reagent_containers/syringe/update_icon(A)
 	cut_overlays()
 
 	if(mode == SYRINGE_BROKEN)
 		icon_state = "broken"
 		return
 
 	var/rounded_vol
-	if(reagents && reagents.total_volume)
-		rounded_vol = CLAMP(round((reagents.total_volume / volume * 15),5), 1, 15)
-		var/image/filling_overlay = mutable_appearance('icons/obj/reagentfillings.dmi', "syringe[rounded_vol]")
-		filling_overlay.color = reagents.get_color()
-		add_overlay(filling_overlay)
-	else
-		rounded_vol = 0
-
-	icon_state = "[rounded_vol]"
-	item_state = "syringe_[rounded_vol]"
-
-	if(ismob(loc))
-		var/injoverlay
-		switch(mode)
-			if (SYRINGE_DRAW)
-				injoverlay = "draw"
-			if (SYRINGE_INJECT)
-				injoverlay = "inject"
-		add_overlay(injoverlay)
-		update_wear_icon()
+	if (sytype == "large")
```

Comment:
It's a wrong way to it. Instead of creating a specialized logic for icon update for each subtype of a syringe, you need to make a base code to be able to handle different kind of overlays instead. Store the overlays as object vars, and use different overlays for child objects without additional procs for them.

Reply:
i was utterly moronic. and now i've got it. thanks for the guidance. """dang it was such a easy fix. and it went right over me...""" well its working now. and as intended